### PR TITLE
Add AddUserAsFriend

### PR DESCRIPTION
**Pull request by shubhamg13**
Please Review.

Just tried to add basic API w/o constraints.

### CODE REVIEW

File: src/main/java/com/n1303/birthdaybuddy/controller/FriendsController.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    @Autowired
+    private FriendService friendService;
+
+    @GetMapping(UrlConstant.ALL_USER_API_URL)
+    public List<Friendship> getAll() {
+        return this.friendService.getAll();
+    }
+
+    @PostMapping(UrlConstant.ADD_FRIENDSHIP)
+    public void insert(@RequestParam("firstUser") String firstUser,
```
Comment (ShervilG) [inline src/main/java/com/n1303/birthdaybuddy/controller/FriendsController.java:24]: Would suggest returning a Friendship here wrapped in ResponseEntity object as done as done in TestController, same for above api.

File: src/main/java/com/n1303/birthdaybuddy/repository/FriendshipRepository.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+    public FriendshipRepository() {
+        this.friendshipMap = new HashMap<>();
+    }
+
+    public List<Friendship> getAll() {
+        return new ArrayList<Friendship>(this.friendshipMap.values());
+    }
+
+    public void insert(String firstUser, String secondUser) {
```
Comment (ShervilG) [inline src/main/java/com/n1303/birthdaybuddy/repository/FriendshipRepository.java:21]: Although it's a test project, It'll be better if we add existence check here before inserting

**Review by ShervilG — CHANGES_REQUESTED** [review]
Good work, left a couple of comments

File: src/main/java/com/n1303/birthdaybuddy/repository/FriendshipRepository.java (hunk 2 above)
Comment (shubhamg13) [reply src/main/java/com/n1303/birthdaybuddy/repository/FriendshipRepository.java:21]: Sure.

File: src/main/java/com/n1303/birthdaybuddy/controller/FriendsController.java (hunk 1 above)
Comment (shubhamg13) [reply src/main/java/com/n1303/birthdaybuddy/controller/FriendsController.java]: Sure

**Review by ShervilG — APPROVED** [review]

**Review by ShervilG — APPROVED** [review]

**Pull request merged**